vizro-core/src/vizro/_vizro.py

```diff
@@ -108,9 +108,10 @@ def build(self, dashboard: Dashboard):
 
         # Note that model instantiation and pre_build are independent of Dash.
         self._pre_build()
-
         self.dash.layout = dashboard.build()
 
+        # Add data-bs-theme attribute for 404 error page
+        self.dash.index_string = self.dash.index_string.replace("<html>", "<html data-bs-theme='dark'")
```

Comment:
```suggestion
        bootstrap_theme = self.dashboard.theme.removeprefix("vizro_")
        self.dash.index_string = self.dash.index_string.replace("<html>", f"<html data-bs-theme='{bootstrap_theme}'")
```

The reason this solution is good is because it hopefully works for both dark and light themes 🙂